plutus-tx-plugin/src/PlutusTx/Compiler/Utils.hs

```diff
@@ -21,14 +22,32 @@ import Language.Haskell.TH.Syntax qualified as TH
 import Data.Map qualified as Map
 import Data.Text qualified as T
 
--- | Get the 'GHC.TyThing' for a given 'TH.Name' which was stored in the builtin name info,
+-- | Get the 'GHC.TyCon' for a given 'TH.Name' stored in the builtin name info,
 -- failing if it is missing.
-getThing :: Compiling uni fun m ann => TH.Name -> m GHC.TyThing
-getThing name = do
-    CompileContext{ccNameInfo=names} <- ask
-    case Map.lookup name names of
-        Nothing    -> throwSd CompilationError $ "Missing name:" GHC.<+> (GHC.text $ show name)
-        Just thing -> pure thing
+lookupGhcTyCon :: Compiling uni fun m ann => TH.Name -> m GHC.TyCon
+lookupGhcTyCon thName = do
+  CompileContext { ccNameInfo } <- ask
+  case Map.lookup thName ccNameInfo of
+    Just (GHC.ATyCon tc) -> pure tc
+    _ -> throwPlain $ CompilationError $ "TyCon not found: " <> T.pack (show thName)
```

Comment:
Use `_` or `Nothing` consistently. Btw these three functions look very similar.

Reply:
In this case `Nothing` couldn't be used because the first pattern is refutable and `Nothing` makes this case non-exhaustive.

Reply:
> Btw these three functions look very similar.

Indeed, I also noticed similarities and considered extracting it, but then after some consideration I decided not to do it because the indirection added seemed not worth reduced repetition to me.